rfc/rfc-100/rfc-100.md

```diff
@@ -0,0 +1,238 @@
+
+2. **RFC-99 BLOB Types**: Introduces BINARY and LARGE_BINARY types to the Hudi type system, providing the type foundation for unstructured data storage.
+
+## Requirements 
+
+Below are the high-level requirements for this feature.
+
+1. Users must be able to define tables with a mix of structured (current types) and unstructured (blob type)
+   columns
+2. Records are distributed across file groups like regular Hudi storage layout into file groups. But within each
+   file group, structured and unstructured columns are split into different column groups. This way the table can
+   also scalably grow in terms of number of columns.
+3. Unstructured data can be stored inline (e.g small images right inside the column group) or out-of-line (e.g
+   pointer to a multi-GB video file someplace). This decision should be made dynamically during write/storage time.
+3. All table life-cycle operations and table services work seamlessly across both column types.for e.g cleaning
+   the file slices should reclaim both inline and out-of-line blob data. Clustering should be able re-organize
+   records across file groups or even redistribute columns across column groups within the same file group.
+4. Storage should support different column group distributions i.e different membership of columns
+   across column groups, across file groups, to ensure users or table services can flexibly reconfigure all this as
+   table grows, without re-writing all of the data.
+5. Hudi should expose controls at the writer level, to control whether new columns are written to new column
+   groups or expand an existing column group within a file group.
+
+
+## High-Level Design
+
+The design introduces a hybrid storage model where each file group can contain multiple column groups with different file formats optimized for their data types. Structured columns continue using 
+Parquet format, while unstructured columns can use specialized formats like Lance or optimized Parquet configurations or HFile for random-access.
+
+### 1. Mixed Base File Format Support
+
+**Per-Column Group Format Selection**: Each column group within a file group can use different base file formats:
+- **Structured Column Groups**: Continue using Parquet with standard optimizations
+- **Unstructured Column Groups**: Use Lance format for vector/embedding data or specially configured Parquet for BLOB storage
+
+**Format Configuration**: File format is determined at column group creation time based on (per the current RFC-80). 
+But, ideally all these configurations should be automatic and Hudi should auto-generate colum group names and mappings.
+
+
+```sql
+CREATE TABLE multimedia_catalog (
+  id BIGINT,
+  product_name STRING,
+  category STRING,
+  image BINARY,
+  video LARGE_BINARY,
+  embeddings ARRAY<FLOAT>
+) USING HUDI
+TBLPROPERTIES (
```

Comment:
Since we know the data types provided in the schema, would it be better user experience for hudi to determine which columns should be stored in a particular column group which maps to the file format suited for that type, rather than the user having to provide this info and the format they want this stored in via table props? 

For large feature tables that may have many columns, this might be cubersome. Having this as optional for configurability makes sense